packages/chain-adapters/src/evm/EVMBaseAdapter.ts

```diff
@@ -0,0 +1,285 @@
+          blockTime: parsedTx.blockTime,
+          chainId: parsedTx.chainId,
+          chain: this.getType(),
+          confirmations: parsedTx.confirmations,
+          txid: parsedTx.txid,
+          fee: parsedTx.fee,
+          status: getStatus(parsedTx.status),
+          tradeDetails: parsedTx.trade,
+          transfers: parsedTx.transfers.map((transfer) => ({
+            assetId: transfer.assetId,
+            from: transfer.from,
+            to: transfer.to,
+            type: getType(transfer.type),
+            value: transfer.totalValue
+          })),
+          data: parsedTx.data
+        }
+      })
+    )
+
+    return {
+      cursor: data.cursor ?? '',
+      pubkey: input.pubkey,
+      transactions: txs
+    }
+  }
+
+  async signTransaction(signTxInput: SignTxInput<ETHSignTx>): Promise<string> {
+    try {
+      const { txToSign, wallet } = signTxInput
+      const signedTx = await (wallet as ETHWallet).ethSignTx(txToSign)
+
+      if (!signedTx) throw new Error('Error signing tx')
+
+      return signedTx.serialized
+    } catch (err) {
+      return ErrorHandler(err)
+    }
+  }
+
+  async signAndBroadcastTransaction(signTxInput: SignTxInput<ETHSignTx>): Promise<string> {
+    try {
+      const { txToSign, wallet } = signTxInput
+      const ethHash = await (wallet as ETHWallet)?.ethSendTx?.(txToSign)
+
+      if (!ethHash) throw new Error('Error signing & broadcasting tx')
+      return ethHash.hash
+    } catch (err) {
+      return ErrorHandler(err)
+    }
+  }
+
+  async broadcastTransaction(hex: string) {
+    const { data } = await this.providers.http.sendTx({ sendTxBody: { hex } })
+    return data
+  }
+
+  async signMessage(signMessageInput: SignMessageInput<ETHSignMessage>): Promise<string> {
+    try {
+      const { messageToSign, wallet } = signMessageInput
+      const signedMessage = await (wallet as ETHWallet).ethSignMessage(messageToSign)
+
+      if (!signedMessage) throw new Error('EthereumChainAdapter: error signing message')
+
+      return signedMessage.signature
+    } catch (err) {
+      return ErrorHandler(err)
+    }
+  }
+
+  async getAddress(input: GetAddressInput): Promise<string> {
+    const { wallet, bip44Params = EVMBaseAdapter.defaultBIP44Params, showOnDevice } = input
+    const path = toPath(bip44Params)
+    const addressNList = bip32ToAddressNList(path)
+    const ethAddress = await (wallet as ETHWallet).ethGetAddress({
+      addressNList,
+      showDisplay: showOnDevice
+    })
+    return ethAddress as string
+  }
+
+  async validateAddress(address: string): Promise<ValidAddressResult> {
+    const chainLabel = chainIdToChainLabel(this.chainId)
+    const isValidAddress = WAValidator.validate(address, chainLabel)
+    if (isValidAddress) return { valid: true, result: ValidAddressResultType.Valid }
+    return { valid: false, result: ValidAddressResultType.Invalid }
+  }
+
+  async validateEnsAddress(address: string): Promise<ValidAddressResult> {
```

Comment:
does this belong in the base class? looks like avalanche has ANS as opposed to ENS, this should prob just exist in the ethereum impl

Reply:
nope good catch!